code/game/objects/items/defib.dm

```diff
@@ -604,7 +604,10 @@
 						var/overall_damage = total_brute + total_burn + H.getToxLoss() + H.getOxyLoss()
 						var/mobhealth = H.health
 						H.adjustOxyLoss((mobhealth - HALFWAYCRITDEATH) * (H.getOxyLoss() / overall_damage), 0)
-						H.adjustToxLoss((mobhealth - HALFWAYCRITDEATH) * (H.getToxLoss() / overall_damage), 0)
+						if(HAS_TRAIT(H, TRAIT_TOXINLOVER))
+							H.adjustToxLoss(-(mobhealth - HALFWAYCRITDEATH) * (H.getToxLoss() / overall_damage), 0)
+						else
+							H.adjustToxLoss((mobhealth - HALFWAYCRITDEATH) * (H.getToxLoss() / overall_damage), 0)
```

Comment:
`adjustToxLoss()` has an arg for making things heal toxin lovers without increasing their blood level, `forced`.
Just calling the proc with a positive value restores their blood level

Reply:
Saw that arg, but wasn't sure if it was the right way to go about it, thanks!